### ISSUE TITLE
Windows Git Bash Vim ALE

### ISSUE THREAD

**Issue by coffeebe4code** [rust-lang/rust-analyzer#18659]
On windows I am in a git bash shell, within vim, using ALE.

I am receiving this error "url is not a file"

**Comment by lnicola** [comment]
Can you patch that to display `url`? But anyway, we should upgrade `lsp-types`, and `Url` has been replaced by `Uri` in the latest version.

**Comment by coffeebe4code** [comment]
If I'm understanding that link. I should be able to downgrade the rust-analyzer version? Which version should I go to?

> Can you patch that to display url?

This will take me some time. I'll see if I can do that sometime today.

**Comment by lnicola** [comment]
> If I'm understanding that link. I should be able to downgrade the rust-analyzer version?

No, I don't think this is a recent issue. The link is more about how we probably can't upgrade `lsp-types` right now because the new version is missing some functionality.

**Comment by coffeebe4code** [comment]
```
url: 'file:///c/Users/ChrisClark/type-lang/linker/src/lib.rs' is not a file
```

**Comment by lnicola** [comment]
Well, that's funny... At least on Linux:

```
[src/main.rs:19:5] Url::parse("file:///c/Users/ChrisClark/type-lang/linker/src/lib.rs").unwrap().to_file_path() = Ok(
    "/c/Users/ChrisClark/type-lang/linker/src/lib.rs",
)
```

**Comment by coffeebe4code** [comment]
I went to the Url crate and found that function, but the line i referenced uses lsp_types::Url, and is calling url.to_file_path()

```
pub(crate) fn abs_path(url: &lsp_types::Url) -> anyhow::Result<AbsPathBuf> {
    let path =
        url.to_file_path().map_err(|()| anyhow::format_err!("url: '{}' is not a file", url))?;
    Ok(AbsPathBuf::try_from(Utf8PathBuf::from_path_buf(path).unwrap()).unwrap())
}
```

**Comment by lnicola** [comment]
`Url` is a re-export in `lsp-types 0.95.0`: https://docs.rs/lsp-types/0.95.0/src/lsp_types/lib.rs.html#26.

**Comment by coffeebe4code** [comment]
I didn't want to spend too much time going through the URL crate. But it annoyed me very quickly.

As a file path, my path is obviously legitimate. I found a few pieces of code that had `cfg!` checks for windows. Which is silly to say the least. I see this a lot across lots of projects. People aren't making their libraries api's cross-platform. I may be on a linux machine, and want to validate if a windows file path is valid.

I'm very interested in developing on Windows, and would be happy to make a change to support it. What is the shortest path that would be acceptable by the rust-analyzer team for me to do this?
